core/src/test/scala/org/apache/spark/CheckpointSuite.scala

```diff
@@ -82,7 +82,7 @@ trait RDDCheckpointTester { self: SparkFunSuite =>
     }
 
     // Test whether dependencies have been changed from its earlier parent RDD
-    assert(operatedRDD.dependencies.head.rdd != parentRDD)
+    assert(operatedRDD.dependencies.head != parentRDD)
```

Comment:
Please fix the original misnomer that encouraged this bad comparison: `parentRDD` is not an RDD, but rather a Dependency.